test/luatest_helpers/server.lua

```diff
@@ -273,4 +346,62 @@ function Server:grep_log(what, bytes, opts)
     return found
 end
 
+-- After this function returns server 100% thinks he owns synchro queue.
+-- If actual promotion happened (election term changed) it returns true,
+-- if term didn't change it return false.
+function Server:promote()
+    self:exec(function(delay, timeout)
+        local old_election_term = box.info.election.term
+        local deadline = require('clock').time() + timeout
+        repeat
+            if require('clock').time() > deadline then
+                error(('Promoting server %d timed out'):format(box.info.id))
+            end
+            pcall(box.ctl.promote)
+            require('fiber').sleep(delay)
```

Comment:
Using fiber.yield() here instead of sleep leads to more tries overall, because promote/demote usually fail because of interrupt from another node and it's better to wait until all interrupting negotiations end. I rewrote those loops and now they are fast enough IMO.